src/arkode/arkode_io.c

```diff
@@ -819,389 +877,220 @@ int arkSetMaxNumConstrFails(void *arkode_mem, int maxfails)
-  > 1.0.  Any illegal value implies a reset to the default.
-  ---------------------------------------------------------------*/
-int arkSetMaxGrowth(void *arkode_mem, realtype mx_growth)
-{
-  int retval;
-  ARKodeHAdaptMem hadapt_mem;
-  ARKodeMem ark_mem;
-  retval = arkAccessHAdaptMem(arkode_mem, "arkSetMaxGrowth",
-                              &ark_mem, &hadapt_mem);
-  if (retval != ARK_SUCCESS)  return(retval);
-
-  /* set allowed value, otherwise set default */
-  if (mx_growth <= ONE) {
-    hadapt_mem->growth = GROWTH;
-  } else {
-    hadapt_mem->growth = mx_growth;
-  }
-
-  return(ARK_SUCCESS);
-}
-
-
-/*---------------------------------------------------------------
-  arkSetMinReduction:
-
-  Specifies the minimum possible step size reduction factor to be
-  allowed between successive integration steps. Allowable values
-  must be > 0.0 and < 1.0. Any illegal value implies a reset to
-  the default.
-  ---------------------------------------------------------------*/
-int arkSetMinReduction(void *arkode_mem, realtype eta_min)
-{
-  int retval;
-  ARKodeHAdaptMem hadapt_mem;
-  ARKodeMem ark_mem;
-  retval = arkAccessHAdaptMem(arkode_mem, "arkSetMinReduction",
-                              &ark_mem, &hadapt_mem);
-  if (retval != ARK_SUCCESS) return(retval);
-
-  /* set allowed value, otherwise set default */
-  if (eta_min >= ONE || eta_min <= ZERO) {
-    hadapt_mem->etamin = ETAMIN;
-  } else {
-    hadapt_mem->etamin = eta_min;
-  }
-
-  return(ARK_SUCCESS);
-}
-
-
-/*---------------------------------------------------------------
-  arkSetFixedStepBounds:
-
-  Specifies the step size growth interval within which the step
-  size will remain unchanged.  Allowable values must enclose the
-  value 1.0.  Any illegal interval implies a reset to the default.
-  ---------------------------------------------------------------*/
-int arkSetFixedStepBounds(void *arkode_mem, realtype lb, realtype ub)
-{
-  int retval;
-  ARKodeHAdaptMem hadapt_mem;
-  ARKodeMem ark_mem;
-  retval = arkAccessHAdaptMem(arkode_mem, "arkSetFixedStepBounds",
-                              &ark_mem, &hadapt_mem);
-  if (retval != ARK_SUCCESS)  return(retval);
-
-  /* set allowable interval, otherwise set defaults */
-  if ((lb <= ONE) && (ub >= ONE)) {
-    hadapt_mem->lbound = lb;
-    hadapt_mem->ubound = ub;
-  } else {
-    hadapt_mem->lbound = HFIXED_LB;
-    hadapt_mem->ubound = HFIXED_UB;
-  }
-
-  return(ARK_SUCCESS);
-}
-
-
-/*---------------------------------------------------------------
-  arkSetAdaptivityMethod:
+  arkSetAdaptivityMethod:  ***DEPRECATED***
 
   Specifies the built-in time step adaptivity algorithm (and
   optionally, its associated parameters) to use.  All parameters
   will be checked for validity when used by the solver.
+
+  Users should transition to constructing non-default SUNControl
+  objects directly, and providing those directly to the integrator
+  via the time-stepping module *SetController routines.
```

Comment:
See earlier comment about deprecating the current signature of the method but switching to a new version later that will allow users to select from the provided controllers without having to construct one themselves.